libmscore/style.cpp

```diff
@@ -2485,6 +2541,251 @@ void MStyle::reset(Score* score)
             score->undo(new ChangeStyleVal(score, st.styleIdx(), MScore::defaultStyle().value(st.styleIdx())));
       }
 
+//--------------------------------------------------------------------------
+//   initPageLayout - only applied to MScore::_baseStyle and _defaultStyle
+//                    must wait until setMscoreLocale() is called in main()
+//--------------------------------------------------------------------------
+
+void MStyle::initPageLayout()
```

Comment:
Here is the only example of an issue that I can see, and it is limited to these circumstances:
1) Your score uses the default page size
2a) Someone else in a different locale opens your score in their default page size, and it does not look the same. It won't look too different because A4 and Letter are similar sizes, but it might break systems differently, or even create extra pages because of overflow.
2b) You toggle your preference to/from mm/inch in the new preferences dialog. The next time you open a default-page-size score (nothing written to the file) it will be in a different page size.

On the other hand, the reason for not storing the default page size is so that MuseScore templates open in the user's default page size, not always in A4.  That's a very positive feature, and should be maintained.

The alternative is to always store the page size, regardless of whether it matches the default.  But that means either templates will all be A4, or there must be a special "save as template" feature that does not store the page size.

Reply:
That's a real concern, though.  Lett and A4 are different enough that it can force measures to different systems & systems to different pages.

We do have special handling to make sure newly created scores are either A4 or Letter depending on locale (actually default printer default paper size, I think).  But when opening an existing score as opposed to creating a new one, it's important to honor the page size.  We do have a template save mode, I think it's a command line option.  In the past we've also just special-cased A4 & Letter when creating from templates, though.